source/Calamari/Kubernetes/ResourceStatus/ResourceStatusCheckTask.cs

```diff
@@ -62,6 +62,20 @@ public class ResourceStatusCheckTask
                     var definedResourceStatuses = resourceRetriever
                                                   .GetAllOwnedResources(definedResources, kubectl, options)
                                                   .ToArray();
+
+                    var nonNamespacedDefinedResources = definedResourceStatuses
+                        .Where(resourceStatus => !resourceStatus.Namespaced)
+                        .Select(resourceStatus => new ResourceIdentifier(
+                            resourceStatus.Kind, 
+                            resourceStatus.Name,
+                            resourceStatus.Namespace))
+                        .ToHashSet();
+
+                    // Filter out cluster-wide resources
+                    definedResources = definedResources
+                        .Where(resource => !string.IsNullOrEmpty(resource.Namespace) && !nonNamespacedDefinedResources.Contains(new ResourceIdentifier(resource.Kind, resource.Name, string.Empty)))
```

Comment:
If either the namespace on the identifier itself is empty, or the namespace from the resource response is empty, we consider it cluster-scoped.